Event/RefreshEvent.php

```diff
@@ -12,27 +12,41 @@
 namespace Gesdinet\JWTRefreshTokenBundle\Event;
 
 use Gesdinet\JWTRefreshTokenBundle\Model\RefreshTokenInterface;
-use Symfony\Component\Security\Guard\Token\PostAuthenticationGuardToken;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
 
 class RefreshEvent extends Event
 {
+    /**
+     * @var RefreshTokenInterface
+     */
     private $refreshToken;
 
-    private $preAuthenticatedToken;
+    /**
+     * @var TokenInterface
+     */
+    private $token;
 
-    public function __construct(RefreshTokenInterface $refreshToken, PostAuthenticationGuardToken $preAuthenticatedToken)
+    public function __construct(RefreshTokenInterface $refreshToken, TokenInterface $token)
     {
         $this->refreshToken = $refreshToken;
-        $this->preAuthenticatedToken = $preAuthenticatedToken;
+        $this->token = $token;
     }
 
     public function getRefreshToken()
     {
         return $this->refreshToken;
     }
 
+    /**
+     * @deprecated use getToken() instead
+     */
     public function getPreAuthenticatedToken()
```

Comment:
Deprecation since the token being retrieved here isn't guaranteed to be a pre-authentication, or post-authentication, token with the interface typehinted.